src/test/java/fi/fmi/avi/archiver/message/populator/ProductMessageTypesValidatorTest.java

```diff
@@ -0,0 +1,88 @@
+package fi.fmi.avi.archiver.message.populator;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.testing.NullPointerTester;
+import fi.fmi.avi.archiver.file.FileMetadata;
+import fi.fmi.avi.archiver.file.InputAviationMessage;
+import fi.fmi.avi.archiver.message.ArchiveAviationMessage;
+import fi.fmi.avi.archiver.message.ProcessingResult;
+import fi.fmi.avi.model.MessageType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+import java.util.Set;
+
+import static fi.fmi.avi.model.MessageType.METAR;
+import static fi.fmi.avi.model.MessageType.SPECI;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ProductMessageTypesValidatorTest {
+
+    private static final String TEST_PRODUCT_IDENTIFIER = "test_product";
+    private static final String ANOTHER_PRODUCT_IDENTIFIER = "another_product";
+    private static final Set<MessageType> TYPE_IDENTIFIERS = ImmutableSet.of(SPECI, METAR);
```

Comment:
Using `MessagePopulatorTests.TYPE_IDS` map, I'd recommend using `MessagePopulatorTests.TypeId` instances for `MessageType` and id number values throughout the test class to ensure test correctness and better readability.

e.g. here

```java
    private static final Set<MessageType> TYPE_IDENTIFIERS = ImmutableSet.of(TypeId.SPECI.getType(), TypeId.METAR.getType());
```

or

```java
    private static final Set<MessageType> TYPE_IDENTIFIERS = Stream.of(TypeId.SPECI, TypeId.METAR)//
            .map(TypeId::getType)//
            .collect(ImmutableSet.toImmutableSet());
```

Reply:
Fixed 👍